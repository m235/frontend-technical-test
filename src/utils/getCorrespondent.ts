import type { Conversation } from '@/types/conversation'

export const getCorrespondent = (userId: number, conversation: Conversation) => {
  switch (userId) {
    case conversation.senderId:
      return conversation.recipientNickname
    case conversation.recipientId:
      return conversation.senderNickname
    default:
      throw new Error('You are nor the recipient nor the sender')
  }
}
