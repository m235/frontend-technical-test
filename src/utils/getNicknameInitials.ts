export const getNicknameInitials = (nickname: string) =>
  nickname
    .split(' ')
    .map((string) => string.trim().charAt(0))
    .slice(0, 2)
    .join('')
