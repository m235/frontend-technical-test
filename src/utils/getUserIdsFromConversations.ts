import type { Conversation } from '@/types/conversation'

export const getUserIdsFromConversations = (conversations?: Conversation[]) =>
  Array.from<number>(new Set(conversations?.reduce((acc, curr) => [...acc, curr.recipientId, curr.senderId], []) ?? []))
