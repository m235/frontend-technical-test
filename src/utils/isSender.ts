import type { Conversation } from '@/types/conversation'

export const isSender = (userId: number, conversation: Conversation) => userId === conversation.senderId
