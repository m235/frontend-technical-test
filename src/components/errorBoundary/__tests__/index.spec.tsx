import { render } from '@/tests/utils'

import ErrorBoundary from '@/components/errorBoundary'

describe('components:errorBoundary', () => {
  const FailingComponent = () => {
    throw new Error('Oops')
  }
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {})
  })
  it('should render default correctly', () => {
    const { container } = render(<ErrorBoundary>Children</ErrorBoundary>)

    expect(container).toMatchSnapshot()
  })

  it('should render the fallback in case of failure', () => {
    const FailingComponent = () => {
      throw new Error('Oops')
    }

    const { getByTestId } = render(
      <ErrorBoundary fallbackComponent={<h1 data-testid="fallback">Error</h1>}>
        <FailingComponent />
      </ErrorBoundary>
    )

    const element = getByTestId('fallback')
    expect(element).toBeTruthy()
    expect(element.textContent).toEqual('Error')
  })

  it('should render nothing if no fallback', () => {
    const { container } = render(
      <ErrorBoundary>
        <FailingComponent />
      </ErrorBoundary>
    )

    expect(container.childElementCount).toEqual(0)
  })
})
