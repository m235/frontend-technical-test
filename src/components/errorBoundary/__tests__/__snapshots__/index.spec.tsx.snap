// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:errorBoundary should render default correctly 1`] = `
<div>
  Children
</div>
`;
